app/controllers/application_controller.rb

```diff
@@ -5,6 +5,139 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get '/' do
+    erb :index
+  end
+
+  get '/signup' do
+    if  logged_in?
+      redirect '/tweets'
+    else
+     erb :signup
+   end
+  end
+
+  post '/signup' do
+    @user = User.new(user_name: params[:user_name],email: params[:email], password: params[:password])
+
+    if @user.save && @user.user_name != "" && @user.email != "" && @user.password != ""
+      session[:user_id] =@user.id
+      redirect '/tweets'
+    else
+      redirect '/signup'
+    end
+  end
+
+  get '/login' do
+    if  logged_in?
+      redirect '/tweets'
+    else
+      erb :login
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by(:user_name => params[:user_name])
+
+    if @user && @user.authenticate(params[:password])
+        session[:user_id] = @user.id
+        redirect "/tweets"
+    else
+        redirect "/login"
+    end
+  end
+
+  get '/logout' do
+    if logged_in?
+      session.clear
+      redirect '/login'
+    else
+      redirect '/'
+    end
+  end
+
+  get '/tweets' do
+    if logged_in?
+      @user = User.find(session[:user_id])
+      erb :tweets
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/new' do
+   if logged_in?
+      erb :new
+    else
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content] != ""
+    @tweet = Tweet.new(:content => params[:content], :user_id => session[:user_id])
+    @tweet.save
+    redirect "/tweets/#{@tweet.id}"
+    else
+      redirect '/tweets/new'
+    end
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
+      erb :show
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find(params[:id])
```

Comment:
Shouldn't you verify a tweet belongs to the `current_user` (and not just that any user is logged in) before allowing them to edit?